packages/medusa/src/services/order-edit.ts

```diff
@@ -548,6 +607,8 @@ export default class OrderEditService extends TransactionBaseService {
     orderEdit.total = totals.total
     orderEdit.difference_due = totals.difference_due
 
+    await this.decorateItemsTotals(orderEdit)
```

Comment:
suggestion:

I am wondering if instead, we should not do something like
```ts
async decorateTotals(orderEdit: OrderEdit): Promise<OrderEdit> {
 const manager = this.transactionManager_ ?? this.manager_
  const { order_id, items } = await this.retrieve(orderEditId, {
    select: ["id", "order_id", "items"],
    relations: ["items", "items.tax_lines", "items.adjustments"],
  })

  const order = await this.orderService_
    .withTransaction(manager)
    .retrieve(order_id, {
      relations: [
        "discounts",
        "discounts.rule",
        "gift_cards",
        "region",
        "items",
        "items.tax_lines",
        "items.adjustments",
        "region.tax_rates",
        "shipping_methods",
        "shipping_methods.tax_lines",
      ],
    })
  const computedOrder = { ...order, items } as Order
  await Promise.all([
    await this.orderService_.withTransaction(manager).decorateTotals(computedOrder),
    await this.orderService_.withTransaction(manager).decorateTotals(order)
  ])

  orderEdit.items = computedOrder.items
  orderEdit.discount_total = computedOrder.discount_total
  orderEdit.gift_card_total = computedOrder.gift_card_total
  orderEdit.gift_card_tax_total = computedOrder.gift_card_tax_total
  orderEdit.shipping_total = computedOrder.shipping_total
  orderEdit.subtotal = computedOrder.subtotal
  orderEdit.tax_total = computedOrder.tax_total
  orderEdit.total = computedOrder.total
  orderEdit.difference_due = computedOrder.total - order.total
  
  return orderEdit
}
```

I ve tested it and it works perfectly with the integration tests.

In the above snippet, we benefit from the order service decorating the totals with the more recent approach and attaching the different totals that are needed. We use it on both the original order and the computed one and in the end, the items and totals are re attached to the order edit, but also, the items totals are directly attached by default.

It would mean removing `decorateItemsTotals` and `getTotals` and updating the request confirmation to use the decorate totals instead.

But I think it makes sense here to have this kind of approach wdyt? @fPolic 

@srindom

Reply:
I am also wondering if we don't need to add `"gift_card_transactions",` to the order relation in the retrieve. It looks like it is missing at the moment

Reply:
LGTM - only argument I can think of against this approach is that, in the long run, as the features both grow, we don't want to couple order edit totals computations with order service (and be cautious when we change something in `orderService` decoration logic to not break order editing) but rather have both services use the totals services independently, but I'd be OK with reimplementing it this way if @srindom agrees.

Reply:
Yeah I was also thinking the same about that. but here we are retrieving the order and just attaching the items. In the end it makes me thoughts that it was making sense to call the order service since we are manipulating an order (even casting as an order) and we are already using it.

Reply:
@adrien2p's approach sounds good to me - also agree with the coupling concern, but think it is fine to revisit at some other time 👍